airflow/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -714,26 +738,41 @@ def execute(self, context: Context):
                     time.sleep(self._attempt_interval)
                 else:
                     raise
+
         if self.deferrable:
-            self.defer(
-                timeout=timedelta(seconds=self.max_attempts * self.poll_interval + 60),
-                trigger=RedshiftDeleteClusterTrigger(
-                    cluster_identifier=self.cluster_identifier,
-                    waiter_delay=self.poll_interval,
-                    waiter_max_attempts=self.max_attempts,
-                    aws_conn_id=self.aws_conn_id,
-                ),
-                method_name="execute_complete",
-            )
+            cluster_state = self.redshift_hook.cluster_status(cluster_identifier=self.cluster_identifier)
+            if cluster_state == "cluster_not_found":
+                self.log.info("Cluster deleted successfully")
+            elif cluster_state in ("creating", "modifying"):
+                raise AirflowException(
+                    f"Unable to delete cluster since cluster is currently in status: {cluster_state}"
+                )
+            else:
```

Comment:
similar comment as above. The else clause could be a broader else to raise an exception for all other states.

Reply:
same above